runtime/src/accounts_db.rs

```diff
@@ -2458,6 +2495,13 @@ impl AccountsDB {
             let pubkey = pubkey_account.0;
             self.accounts_index
```

Comment:
nits: maybe add comment here like this?

```rust
// we don't atomically update both primary index and secondary index together.
// This certainly creates small time window with inconsistent state across the two indexes.
// But, this is acceptable because:
//  - we don't care about strict consistent view at any given moment of time because only working 
//     bank's slot will suffer from this. we expects rooted slot is used more often
//  - secondary index is always used while being dependant on the primary index by means of
//      post-filtering via the primary index, not used by its own. So, what accounts_index sees is
//      enough for us to regard as the source of truth.
```

Reply:
That's a really good point. Also scans are only supported on `frozen` banks, so if somebody is trying to do a scan on a working bank, it is not required to be consistent.